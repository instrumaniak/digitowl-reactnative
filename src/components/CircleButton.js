/**
 *   Floating Circle button
 *   used in Dashboard
 *   for create new entry
 */

import React from 'react'
import { View, TouchableOpacity, StyleSheet } from 'react-native'
import { withNavigation } from 'react-navigation'
import { Icon } from 'native-base'
import { colors, themeColors } from '../config'

const CircleButton = ({ navigation }) => (
  <TouchableOpacity
    style={ styles.container }
    onPress={() => navigation.navigate('Create')}>
      <Icon name='md-add' style={{
        fontSize: 30,
        color: colors.whiteSmoke
      }} />
  </TouchableOpacity>
)

const styles = StyleSheet.create({
  container: {
    width: 60,
    height: 60,
    borderRadius: 30,
    backgroundColor: themeColors.brandLighten,
    justifyContent: 'center',
    alignItems: 'center',
    elevation: 10,
    position: 'absolute',
    bottom: 10,
    right: 15
  }
})

export default withNavigation(CircleButton)
