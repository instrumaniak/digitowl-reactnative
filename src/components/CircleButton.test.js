/**
 *   Tests for component: CircleButton
 */

import React from 'react'
import CircleButton from './CircleButton'

import renderer from 'react-test-renderer'

test('Renders correctly', () => {
  const tree = renderer.create(<CircleButton />).toJSON()
  expect(tree).toMatchSnapshot()
})
