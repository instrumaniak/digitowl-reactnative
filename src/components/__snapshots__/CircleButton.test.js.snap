// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Renders correctly 1`] = `
<View
  style={
    Object {
      "alignItems": "center",
      "backgroundColor": "#00b175",
      "borderRadius": 30,
      "bottom": 10,
      "elevation": 10,
      "height": 60,
      "justifyContent": "center",
      "position": "absolute",
      "right": 10,
      "width": 60,
    }
  }
>
  <Text
    allowFontScaling={false}
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 12,
        },
        Array [
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#000",
            "fontSize": 30,
          },
          Object {
            "color": "#F5F3F5",
            "fontSize": 30,
          },
        ],
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
        Object {},
      ]
    }
  >
    ï‰³
  </Text>
</View>
`;
